### PR TITLE
Update EdgeConnect samples 

### PR DESCRIPTION
**Pull request by StefanHauth**
## Description

Update samples for EdgeConnect:
* switch API version
* add sample for K8s automation
* add sample for provisioner mode

## How can this be tested?

not. File review will suffice.

### CODE REVIEW

File: assets/samples/edgeconnect/automation-workflows.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  # The service account has to have the proper permissions that match the purpose of the
+  # workflow steps this EdgeConnect is used for.
+  serviceAccountName: edgeconnect-automation
+
+  # Enable K8S Automation for workflows
+  kubernetesAutomation:
+    enabled: true
+
+  oauth:
+    # OAuth client has to be created in Dynatrace account management
```
Comment (chrismuellner) [inline assets/samples/edgeconnect/automation-workflows.yaml:19]: can we link to docs for this?

File: assets/samples/edgeconnect/provisioner.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+kind: EdgeConnect
+metadata:
+  name: edge-connect-sample
+  namespace: dynatrace
+spec:
+  apiServer: "https://example.dev.apps.dynatracelabs.com"
+  replicas: 1
+
+  oauth:
+    # OAuth client has to be created in Dynatrace account management
```
Comment (chrismuellner) [inline assets/samples/edgeconnect/provisioner.yaml:11]: same here, I'm assuming there's a guide somewhere for this

**Review by chrismuellner — CHANGES_REQUESTED** [review]

File: assets/samples/edgeconnect/provisioner.yaml (hunk 2 above)
Comment (andriisoldatenko) [reply assets/samples/edgeconnect/provisioner.yaml:11]: note sure but found something similar https://docs.dynatrace.com/docs/manage/configuration-as-code/terraform/guides/create-oauth-client

Comment (StefanHauth) [reply assets/samples/edgeconnect/provisioner.yaml:11]: Yeah, there is no "general" description, but there are a bunch of paragraphs that can be helpful.

File: assets/samples/edgeconnect/automation-workflows.yaml (hunk 1 above)
Comment (StefanHauth) [reply assets/samples/edgeconnect/automation-workflows.yaml:22]: 8bb546cca5e0c5f0c601d9eb380c4c95d6e0349e

File: assets/samples/edgeconnect/provisioner.yaml (hunk 2 above)
Comment (StefanHauth) [reply assets/samples/edgeconnect/provisioner.yaml:11]: 8bb546cca5e0c5f0c601d9eb380c4c95d6e0349e

File: assets/samples/edgeconnect/automation-workflows.yaml (hunk 1 above)
Comment (chrismuellner) [reply assets/samples/edgeconnect/automation-workflows.yaml:22]: please remove the list of scopes from the sample. referencing them in the docs is enough

File: assets/samples/edgeconnect/automation-workflows.yaml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    #
+    # required scopes:
+    # - app-engine:edge-connects:connect
+    # - app-engine:edge-connects:read
+    # - app-engine:edge-connects:write
+    # - app-engine:edge-connects:delete
+    # - oauth2:client:manage
+    # - settings:objects:read
+    # - settings:objects:write
+    # - settings:schemas:read
```
Comment (andriisoldatenko) [inline assets/samples/edgeconnect/automation-workflows.yaml:33]: do we really need it? I havent seen it in code, for me it works without this one `settings:schemas:read`

File: assets/samples/edgeconnect/automation-workflows.yaml (hunk 1 above)
Comment (chrismuellner) [reply assets/samples/edgeconnect/automation-workflows.yaml:22]: we don't have the scopes as part of the dynakube api token, so we should not have them for edgeconnect either. the place to document how to create the token and what it should include is in the official documentation, not as part of the example yaml.

Comment (StefanHauth) [reply assets/samples/edgeconnect/automation-workflows.yaml:22]: For what it's worth, I find it more convenient from user PoV. But I've instead linked to the resp. docu page to make it very clear, which scopes are used.

Comment (StefanHauth) [reply assets/samples/edgeconnect/automation-workflows.yaml:22]: 1978807a

File: assets/samples/edgeconnect/provisioner.yaml (hunk 2 above)
Comment (StefanHauth) [reply assets/samples/edgeconnect/provisioner.yaml:14]: 1978807a

**Review by chrismuellner — APPROVED** [review]

**Pull request merged**